### PR TITLE
Add tests for creating vouchers

### PR DESCRIPTION
**Pull request by karola312**
I want to merge this change because I added tests for creating vouchers

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-5615 Add tests for creating vouchers](https://app.clickup.com/t/2549495/SALEOR-5615)

**Review by orzechdev — APPROVED** [review]

File: cypress/integration/discounts/vouchers/vouchers.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            .wait("@VoucherDelete");
+          createCheckoutForCreatedVoucher(voucher.code);
+        })
+        .then(({ addPromoCodeResp }) => {
+          const errorField = addPromoCodeResp.checkoutErrors[0].field;
+          expect(errorField).to.be.eq("promoCode");
+        });
+    });
+
+    function createCheckoutForCreatedVoucher(voucherCode) {
```
Comment (eCzerniak) [inline cypress/integration/discounts/vouchers/vouchers.js:90]: Please move this function to other file.

File: cypress/integration/discounts/vouchers/vouchers.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            product = productResp;
+            createChannel({ name });
+          }
+        )
+        .then(channel => {
+          createdChannel = channel;
+        });
+    });
+
+    it("should delete voucher", () => {
```
Comment (eCzerniak) [inline cypress/integration/discounts/vouchers/vouchers.js:59]: Also I wonder why we have double tests for delete voucher. 
We have this test here and   https://github.com/saleor/saleor-dashboard/pull/1817/files 
 -> cypress/integration/discounts/vouchers/updateVouchers.js

**Review by eCzerniak — CHANGES_REQUESTED** [review]

File: cypress/integration/discounts/vouchers/vouchers.js (hunk 1 above)
Comment (karola312) [reply cypress/integration/discounts/vouchers/vouchers.js:90]: It is fixed in https://github.com/saleor/saleor-dashboard/pull/1817/files

File: cypress/integration/discounts/vouchers/vouchers.js (hunk 2 above)
Comment (eCzerniak) [reply cypress/integration/discounts/vouchers/vouchers.js:59]: @karola312 I did not mention about  cypress/integration/discounts/vouchers/vouchers.js 
I mention about `updateVoucher.js`. There is a difference between `voucher.js` vs `updateVoucher.js`

**Review by eCzerniak — CHANGES_REQUESTED** [review]

File: cypress/integration/discounts/vouchers/vouchers.js (hunk 2 above)
Comment (eCzerniak) [reply cypress/integration/discounts/vouchers/vouchers.js:59]: image

Comment (karola312) [reply cypress/integration/discounts/vouchers/vouchers.js:59]: So if it is about updateVoucher from next PR, I will take care of it in next PR

**Comment by karola312** [comment]
So if it is about updateVoucher from next PR, I will take care of it in next PR

File: cypress/integration/discounts/vouchers/createVouchers.js [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        })
+        .then(({ addPromoCodeResp }) => {
+          const errorField = addPromoCodeResp.checkoutErrors[0].field;
+          expect(errorField, "error in promo code should occur").to.be.eq(
+            "promoCode"
+          );
+        });
+    });
+
+    it("should be able to create voucher with limit to one use per customer. TC: SALEOR_1908", () => {
```
Comment (eCzerniak) [inline cypress/integration/discounts/vouchers/createVouchers.js:211]: Why we do not check in first checkout if promoCode is added and there is no error and order is created with discount and why there are 3 createCheckout?

**Review by eCzerniak — CHANGES_REQUESTED** [review]

File: cypress/integration/discounts/vouchers/createVouchers.js (hunk 3 above)
Comment (karola312) [reply cypress/integration/discounts/vouchers/createVouchers.js:212]: 3rd checkout is to check if other (not logged in) customer still can use that voucher in checkout. I added comment in this TC. I added check if there is no errors in first checkout.

File: cypress/integration/discounts/vouchers/vouchers.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import {
+  createVoucherInChannel,
+  deleteVouchersStartsWith
+} from "../../../support/api/utils/discounts/vouchersUtils";
+import { createCheckoutWithVoucher } from "../../../support/api/utils/ordersUtils";
+import * as productsUtils from "../../../support/api/utils/products/productsUtils";
+import filterTests from "../../../support/filterTests";
+
+filterTests({ definedTags: ["all"] }, () => {
+  describe("Vouchers discounts", () => {
```
[the remark this answers is not in the document]
Comment (karola312) [reply cypress/integration/discounts/vouchers/vouchers.js:19]: It will be changed in next PR

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**